samcli/scripts/copy_terraform_built_artifacts.py

```diff
@@ -0,0 +1,215 @@
+        for token in self.tokens:
+            self.resolvers.append(self.find_resolver(token))
+
+    def parse(self):
+        """
+        Instantiate a searcher that returns parsed data based on the resolvers.
+        :return:
+        """
+        return Searcher(self.resolvers)
+
+    def find_resolver(self, token):
+        """
+        Find the resolver for the appropriate token. The implementation of this function
+        is a direct match against a regex as the number of uses-case to be supported are less.
+        """
+        groups = self.list_resolver_regex.findall(token)
+        if not groups:
+            return KeyResolver(key=token)
+        else:
+            return ListConditionResolver(key=groups[0][0], value=groups[0][1])
+
+
+class Searcher(object):
+    """
+    Searcher class that allows for searching a Jpath against structured data.
+    """
+
+    def __init__(self, resolvers):
+        self.resolvers = resolvers
+
+    def search(self, data):
+        """
+        Search by applying all resolvers against structured data.
+        """
+
+        for resolver in self.resolvers:
+            data = resolver.resolve(data)
+        return data
+
+
+if __name__ == "__main__":
+    # Gather inputs and clean them
+    argparser = argparse.ArgumentParser(
+        description="Copy built artifacts referenced in a json file " "(passed via stdin) matching a search pattern"
+    )
+    argparser.add_argument(
+        "--expression",
+        type=str,
+        required=True,
+        help="Jpath query expression separated by | (delimiter) "
+        "and allows for searching within a json object."
+        "eg: |values|sub_module|[?address==me]|output",
+    )
+    argparser.add_argument(
+        "--directory",
+        type=str,
+        required=True,
+        help="directory to which extracted expression " "contents are copied/unzipped to",
+    )
+
+    arguments = argparser.parse_args()
+    directory_path = os.path.abspath(arguments.directory)
+
+    if not os.path.exists(directory_path) or not os.path.isdir(directory_path):
+        raise OSError("Expected --directory-path to be a valid directory")
+
+    # Load and Parse
+    data_object = json.load(sys.stdin)
+    output_path = Parser(expression=arguments.expression).parse().search(data=data_object)
+
+    # Check if that is indeed a path.
+    if not os.path.exists(output_path):
+        raise OSError("Extracted attribute path from provided expression does not exist!")
+    filepath = os.path.abspath(output_path)
+
+    # Unzip the zipped file or copy the dir
+    try:
+        if zipfile.is_zipfile(filepath):
+            with zipfile.ZipFile(filepath, "r") as z:
+                z.extractall(directory_path)
+        else:
+            if os.path.isdir(directory_path):
+                shutil.copytree(output_path, directory_path, dirs_exist_ok=True)
+    except OSError:
+        print("Copy/Unzip unsuccessful!")
```

Comment:
can we log the root cause exception, it will be helpful in debugging customers' issues.